oShareTests/XCTest+Eventually.swift

```diff
@@ -0,0 +1,41 @@
+import XCTest
+
+extension XCTestCase {
+	
+	/// Simple helper for asynchronous testing.
+	/// Usage in XCTestCase method:
+	///   func testSomething() {
+	///       doAsyncThings()
+	///       eventually {
+	///           /* XCTAssert goes here... */
+	///       }
+	///   }
+	/// Cloure won't execute until timeout is met. You need to pass in an
+	/// timeout long enough for your asynchronous process to finish, if it's
+	/// expected to take more than the default 0.01 second.
+	///
+	/// - Parameters:
+	///   - timeout: amout of time in seconds to wait before executing the
+	///              closure.
+	///   - closure: a closure to execute when `timeout` seconds has passed
+	func eventually(timeout: TimeInterval = 0.01, closure: @escaping () -> Void) {
+		let expectation = self.expectation(description: "")
+		expectation.fulfillAfter(timeout)
+		self.waitForExpectations(timeout: 60) { _ in
+			closure()
+		}
+	}
+}
+
+extension XCTestExpectation {
+	
+	/// Call `fulfill()` after some time.
+	///
+	/// - Parameter time: amout of time after which `fulfill()` will be called.
+	func fulfillAfter(_ time: TimeInterval) {
+		DispatchQueue.main.asyncAfter(deadline: .now() + time) {
+			self.fulfill()
+		}
+	}
+}
+
```

Comment:
Due to my propensity for using Quick and Nimble in my own projects, I grew used to using `waitUntil()` for things when I wasn't sure how long things will take to complete. I decided to adopt this wrapper which I found a while ago, for use with XCTest.

I did not write the original code for this. Alas I can't seem to find where the original is from. I did update some syntax though.